pkg/cmd/codespace/common.go

```diff
@@ -61,6 +68,10 @@ func (a *App) StopProgressIndicator() {
 	a.io.StopProgressIndicator()
 }
 
+func (a *App) showStatus(ctx context.Context, codespace *api.Codespace) error {
+	return a.statusChecker(a, ctx, codespace)
```

Comment:
this seems like a smell where the method is on the `a *App` but we also have to pass it in as the first argument.
It feels a bit like we are tying ourself in a knot.

Reply:
I agree and I'm also not a fan. Apart from ripping some things apart, though, I'm not sure how to improve it. Any other ideas? I think this is all the app is used for, so maybe it wouldn't be too disruptive to refactor it in this PR.

https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/pkg/cmd/codespace/create.go#L349
https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/pkg/cmd/codespace/create.go#L358

Reply:
Maybe add a new `progressIndicator` interface?

```go
type progressIndicator interface {
  StartWithLabel(string)
  Stop()
}
```

The App is basically delegating this to the `io iostreams.IOStreams` field, so we could pass the `io` in instead of the `app`?

https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/pkg/cmd/codespace/common.go#L54-L62

Reply:
Yeah extracting the interface would be preferable, but it might be too many changes to make as part of this PR? I think it would touch a lot of files. But yeah I def think we should try to clean this up in the future

Reply:
I gave it a shot in https://github.com/cli/cli/pull/5753/commits/cea1dff37423357ea15194b9ef829fc28cab7d2e

It's not too bad, but it's also hitting up against the somewhat arbitrary-feeling divide between the `codespaces` and `codespace` packages. 🤷🏻 

Let me know what you think

Reply:
yeah I think this is a good first step at cleaning this up! I'm gonna work on untangling the `codespace` + `codespaces` packages into a single package. which should allow us to cleanup the codebase even further, but its probably better done in a separate PR

Reply:
I think that would be valuable. Let me know how I can help